regression_detector.py

```diff
@@ -158,6 +159,31 @@ def process_regressions_into_gh_issue(regressions_dict, owner: str, output_path:
         f.write(issue_body)
 
 
+def get_best_start_date(latest_metrics_jsons, end_date: str) -> Optional[datetime]:
+    """Get the date closest to `end_date` from `latest_metrics_jsons`"""
+    end_datetime = datetime.strptime("%Y-%m-%d", end_date)
+    for metrics_json in latest_metrics_jsons:
+        start_datetime = datetime.strptime(metrics_json.split('/')[-1].split('-')[-1].split('.')[0], '%Y%m%d%H%M%S')
+        if start_datetime < end_datetime:
+            return start_datetime
+    return None
+
+
+def get_metrics_by_date(latest_metrics_jsons: List[str], pick_date: datetime):
+    pick_metrics_json_key: Optional[str] = None
+    for metrics_json in latest_metrics_jsons:
+        metric_datetime = datetime.strptime(metrics_json.split('/')[-1].split('-')[-1].split('.')[0],
+                                                            '%Y%m%d%H%M%S')
+        # Use the latest metric file on on the same day
+        if metric_datetime.date() == pick_date.date():
+            pick_metrics_json_key = metrics_json
+    assert pick_metrics_json_key, f"Selectd date {pick_date} is not found in the latest_metrics_jsons: {latest_metrics_jsons}"
+    s3 = S3Client(USERBENCHMARK_S3_BUCKET, USERBENCHMARK_S3_OBJECT)
+    metrics_json_file = s3.get_file_as_json(pick_metrics_json_key)
```

Comment:
This should just be a json already and not a file

Reply:
Nice catch! Thanks!